WordPress/src/main/java/org/wordpress/android/GCMIntentService.java

```diff
@@ -126,10 +132,22 @@ private void handleDefaultPush(Context context, Bundle extras) {
         mPreviousNoteId = noteId;
         mPreviousNoteTime = thisTime;
 
-        // Get a unique Id per notification
-        int pushId = PUSH_NOTIFICATION_ID + mActiveNotificationsMap.size();
+        int pushId = 0;
+
+        // Update notification content for the same noteId if it is already showing
+        boolean matchedNoteId = false;
+        for (int id : mActiveNotificationsMap.keySet()) {
+            Bundle noteBundle = mActiveNotificationsMap.get(id);
+            if (noteBundle.getString(PUSH_ARG_NOTE_ID, "").equals(noteId)) {
+                pushId = id;
+                mActiveNotificationsMap.put(pushId, extras);
+                matchedNoteId = true;
+                break;
+            }
+        }
 
-        if (noteId != null && !mActiveNotificationsMap.containsKey(noteId)) {
+        if (!matchedNoteId) {
```

Comment:
The variable `matchedNoteId` isn't really necessary since we can simply check whether `pushId` is still zero.